packages/firestore/src/local/memory_query_cache.ts

```diff
@@ -182,7 +182,7 @@ export class MemoryQueryCache implements QueryCache {
     transaction: PersistenceTransaction,
     query: Query
   ): PersistencePromise<QueryData | null> {
-    const queryData = this.queries.get(query) || null;
+    const queryData = this.queries.getWithoutValueEqualityCheck(query) || null;
```

Comment:
I think we want to undo this but add an assert that query.isTargetQuery is true...